### ISSUE TITLE
"ERROR: 38:0: Missing symbol ;" or malformed JSON when no extra newline

### ISSUE THREAD

**Issue by noizex** [baszalmstra/header-parser#7]
When the file does not have extra newline the parser will error in two ways, depending if the last character is `;` or not. 

**It's important to delete any extra newline so the file ends at the last character of the code**

1) No newline, file ends at `};` -> Causes "ERROR: 38:0: Missing symbol ;"
```
TCLASS()
class FooTest
{
};
```

2) No newline, file ends with `}` -> Causes JSON output to be malformed
```
namespace Foo
{
TCLASS()
class FooTest
{
};
}
```

it would abruptly end the output of JSON like this:
```
[
    {
        "type": "namespace",
        "name": "Foo",
        "members": [
            {
                "type": "class",
                "line": 5,
                "meta": {},
                "name": "FooTest",
                "members": []
            }
        ]
```

Adding special newline after end of code solves this, but if this can be fixed it will reduce need for additional 'pre-processing' of files.

**Comment by noizex** [comment]
PS. The 2) behaviour I noticed before latest fixes - just forgot to report it - so it can't be caused by the EOF fix I think. I found some of my files would end abruptly with malformed JSON and it was thrown from my Python processing script - I ignored them for now as they did not contain my special macros - so another hint is this happens despite macros being or not being in the file - it would not close the final bracket for the array when only dumping the includes.
